api/v1alpha1/nutanix_node_types.go

```diff
@@ -60,6 +60,24 @@ type NutanixMachineDetails struct {
 	// The project must already be present in the Prism Central.
 	// +kubebuilder:validation:Optional
 	Project *NutanixResourceIdentifier `json:"project,omitempty"`
+
+	// List of GPU devices that need to be added to the machines.
+	// +kubebuilder:validation:Optional
+	GPUs []NutanixGPU `json:"gpus,omitempty"`
+}
+
+type NutanixGPU struct {
+	// Type is the identifier type to use for this resource.
+	// +kubebuilder:validation:Required
+	Type NutanixGPUIdentifierType `json:"type"`
```

Comment:
The CAPX does have NutanixGPUIdentifierType in their schema. however I dont see that it is being used in the code to determine the type. in CAPX logic, the DeviceID takes precedence over name if both specified. 
https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/blob/main/controllers/helpers.go#L687

@deepakm-ntnx @thunderboltsid can you please provide more information if `NutanixGPUIdentifierType` is useful or going to be useful in future and if it make sense to provide it here.

I also like to consider if 
1) Should we only accept Gpu name `name` as input. 
2) if we are have concern about having duplicates in gpu names (I am not sure if its possible), then we can allow both `DeviceID` and `Name` and have CAPX take care of it.

Reply:
2.a. CAREN can also validate to make sure either `name` or `DeviceID` provided.

Reply:
I just noticed. The `Type` is mandatory in the API. https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/blob/main/api/v1beta1/nutanix_types.go#L89C29-L89C37
Please scratch suggestion about not providing type. however we should discuss from CAPX perspective, it we need it in their API

Reply:
Great observation!

If `DeviceID` is not necessary or useful, let's remove it.

Reply:
from my understanding the createVM call needs an UUID so the user can provide a UUID or a name and if they provide the name that function gets the ID for the name

Reply:
Conclusion from discussion: We will keep both fields DeviceID and Name. The main reasons.
1) There can be duplicate names as they are allowed at infrastructure level. DeviceID will present uniqueness.
2) Provide greater flexibility in the API.